include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -747,14 +774,19 @@ __parallel_transform_reduce_then_scan(oneapi::dpl::__internal::__device_backend_
     const std::size_t __n = __in_rng.size();
     const std::size_t __max_inputs_per_block = __work_group_size * __max_inputs_per_item * __num_work_groups;
     std::size_t __num_remaining = __n;
+    if constexpr (__is_unique_pattern_v)
+    {
+        // skip scan of zeroth element in unique patterns
+        __num_remaining -= 1;
```

Comment:
Should we `assert(__num_remaining > 0)` before this to make sure we are properly handling the `__n == 0` case everywhere?

Reply:
Yes, this is a good idea to add after this constexpr block, because the kernel itself is not written to withstand an "empty" scan.

Reply:
done